### ISSUE TITLE
Donate repo to kubernetes-sigs

### ISSUE THREAD

**Issue by ArangoGutierrez** [kubeflow/mpi-operator#459]
TLDR from : #341 

---
There is some renewed interest in moving the mpi-operator to (possibly) kubernetes-sigs, assuming that SIG Apps is willing to sponsor the repo.

The motivation is to encourage non-training users (like HPC) to use and contribute to it, without having to install or learn about kubeflow's training-operator.

- Most Kubeflow components are "loosely coupled", so the definition of "being part of kubeflow" is mostly about if a Kubeflow Working Group is responsible for it

- The committers of the mpi-operator can decide where the best home for it is, but being under the Kubeflow brand certainly will bring more attention to the tool
---

from https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md#rules-for-donated-repositories

- [ ] All contributors must have signed the [CNCF Individual CLA](https://github.com/cncf/cla/blob/master/individual-cla.pdf) or [CNCF Corporate CLA](https://github.com/cncf/cla/blob/master/corporate-cla.pdf)
- [ ] If (a) contributor(s) have not signed the CLA and could not be reached, a NOTICE file should be added referencing section 7 of the CLA with a list of the developers who could not be reached
- [ ] Licenses of dependencies are acceptable; project owners can ping [@caniszczyk](https://github.com/caniszczyk) for review of third party deps
- [ ] Boilerplate text across all files should attribute copyright as follows: "Copyright <Project Authors>" if no CLA was in place prior to donation
- [ ] Additions of [the standard Kubernetes header](https://git.k8s.io/kubernetes/hack/boilerplate/boilerplate.go.txt) to code created by the contributors can occur post-transfer, but should ideally occur shortly thereafter.
- [ ] Should contain template files as per the [kubernetes-template-project](https://github.com/kubernetes/kubernetes-template-project).
- [ ] Kubeflow plan to migrate MPi-Jobs dependencies to the news k8s-sigs/mpi-operator location

After Donation / repo creation under k8s-sigs/mpi-operator
we need to :

- [ ] Stabilize the current code under k8s-gis parameters 
- [ ] Cut a first release under k8s-sigs
- [ ] Migrate Kubeflow projects to new location

Repo will be sponsored by sig-Apps

**Comment by alculquicondor** [comment]
Can you add the rationale for the donation in the description?

Also it would be good to add what's the high-level plan for kubeflow to use the kubernetes-sig/mpi-operator in their manifests.

**Comment by jbottum** [comment]
Perhaps this data point is relevant from the Kubeflow User Survey, which shows MPI's position to other technologies used by Kubeflow users.
<img width="754" alt="Screen Shot 2022-03-09 at 3 08 46 PM" src="https://user-images.githubusercontent.com/10553232/157536808-bf0549e2-113f-49d6-abaa-cc3418336c11.png">

**Comment by ArangoGutierrez** [comment]
Step one: 
- To the question: Are you ok with donating the repo mpi-operator to kubernetes-sigs, please reply lgtm

From the OWNERS file:
@ rongou @terrytangyuan @alculquicondor @carmark @gaocegege @zw0610

**Comment by theadactyl** [comment]
Hi there, @kubeflow/project-steering-group will need to see a proposal with the following information. This should also be open for broader community feedback before any decision is made. 

- Goals
- Motivation / Rationale
- User Benefit 
- Design Proposal
 - Including: 
  - Alternatives Considered
  - Dependencies
  - User Impact
- Detailed Design

**Comment by ArangoGutierrez** [comment]
> Hi there, @kubeflow/project-steering-group will need to see a proposal with the following information. This should also be open for broader community feedback before any decision is made.
> 
> * Goals
> * Motivation / Rationale
> * User Benefit
> * Design Proposal, including:
>   -- Alternatives Considered
>   -- Dependencies
>   -- User Impact
> * Detailed Design

Will work on it, thanks for the check list

**Comment by theadactyl** [comment]
The best place to submit is as a PR to the following directory: https://github.com/kubeflow/community/tree/master/proposals

Thank you! Our proposal process is not widely known, but it's important to follow for any proposed changes that include code governance.

**Comment by rongou** [comment]
lgtm

**Comment by terrytangyuan** [comment]
LGTM.

**Comment by gaocegege** [comment]
/cc @richardsliu 

LGTM

**Comment by carmark** [comment]
LGTM

**Comment by thesuperzapper** [comment]
I also want to highlight that by definition, if a project is donated from Kubeflow to another organization, Kubeflow will no longer have any control over it.

This means there would be a question of what to do with the Kubeflow Working Group which controls that project.

In the case that only one part of a WG is donated, then that WG would continue to exist. 
(Assuming the community thinks there are enough meaningful things that remain under the WG's control)

In the case that most or all of a WG's code is donated, obviously that WG would be disestablished. In such cases, it may also make sense to create a SIG for users of Kubeflow AND that tool (just like we have for other projects that were developed outside of Kubeflow).

**Comment by ArangoGutierrez** [comment]
https://github.com/kubeflow/community/pull/557

**Comment by kannon92** [comment]
Is this still on the table?

**Comment by terrytangyuan** [comment]
There were some additional discussions in https://github.com/kubeflow/community/pull/557 

Now that Kubeflow is an incubating project in CNCF, we need to re-evaluate whether there's enough motivation to move this to K8s as a subproject.

**Comment by alculquicondor** [comment]
I don't think so.

It would be better to just lean on this proposal instead: https://github.com/kubernetes/kubernetes/issues/124537

Note that there is an item in the agenda for it in the next WG training meeting May 1st at 2pm UTC.

And yesterday's recording for k8s WG Batch https://youtu.be/COzNG8XCGTw?si=ncemR5-gmNE3W0OH&t=938

**Comment by kannon92** [comment]
Should we close and reopen if the discussion happens again?
